diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter6/classes/SingleInitRule.kt

```diff
@@ -88,6 +88,14 @@ class SingleInitRule(configRules: List<RulesConfig>) : DiktatRule(
                     .statements
                     .mapNotNull { it.asAssignment() }
                     .filter { it.left is KtNameReferenceExpression }
+                    .filter { state ->
+                        val assignedRef = state.right
+                        if (assignedRef is KtNameReferenceExpression) {
+                            properties.find { (it.psi as KtProperty).name == assignedRef.getReferencedName() } != null
```

Comment:
So, we are filtering out assignments, which use local properties from `init` block in their right hand side, right? This makes sense. I think we should also check more deeply nested references here, e.g. by using `findDescendantsOfType`

Reply:
Fixed